pypower/set_reorder.py

```diff
@@ -20,6 +20,7 @@ def set_reorder(A, B, idx, dim=0):
     """
     A = A.copy()
     ndims = ndim(A)
+    A = A.astype(B.dtype)
```

Comment:
A and B must be the same type. Eg. when B is a complex matrix and the type of A is float, without this code the imag of B will be discarded.